edx_event_bus_kafka/internal/producer.py

```diff
@@ -157,6 +170,41 @@ def inner_to_dict(event_data, ctx=None):  # pylint: disable=unused-argument
     return key_serializer, value_serializer
 
 
+@attr.s(kw_only=True, repr=False)
+class ProducingContext:
+    """
+    Wrapper class to allow us to link a call to produce() with the on_event_deliver callback
+    """
+    full_topic = attr.ib(type=str, default=None)
+    event_type = attr.ib(type=str, default=None)
+    event_key = attr.ib(type=str, default=None)
+    signal = attr.ib(type=OpenEdxPublicSignal, default=None)
+    initial_topic = attr.ib(type=str, default=None)
+    event_key_field = attr.ib(type=str, default=None)
+    event_data = attr.ib(type=dict, default=None)
+
+    def __repr__(self):
+        """Create a logging-friendly string"""
+        return " ".join([f"{key}={value!r}" for key, value in attr.asdict(self).items()])
```

Comment:
Could we just omit this method and drop the `repr=False`?

Reply:
Hmm... I tried this locally and got a test failure. In one of the unit tests, the expected substring `event_data={'test_data': {'sub_name': 'name', 'course_id': 'id'}}` became `event_data={'test_data': SubTestData0(sub_name='name', course_id='id')}` and I'm not sure why. That test-data object should be getting repr'd either way...

Reply:
Weird:
```
(Pdb) attr.asdict(self)['event_data']
{'test_data': {'sub_name': 'name', 'course_id': 'ABCx'}}
(Pdb) self.event_data
{'test_data': SubTestData0(sub_name='name', course_id='ABCx')}
```
OK, turns out that `asdict` recurses into values by default and converts them too! Not what I would have expected at all, and I don't think we want that? https://www.attrs.org/en/stable/api.html#attr.asdict

Reply:
I'm not convinced it matters either way. Also I like having the "=" sign in there so it's more in line with normal splunk logging conventions.

Reply:
Allowing attrs to define `__repr__` for us would mean having less code, but I don't feel strongly about it.

But if we keep it I think I'd like to include `recurse=False` in the call to `asdict` since as it is we're losing type information that could be important for debugging.

Reply:
I feel strongly about the = thing, but I'll add the recurse=False bit

Reply:
[fixed]